Fantom-foundation/Tosca
# CT/LFVM: Memory/Gas overflow checks.

Thanks to the fuzzer continously finding more overflow checks, we digged dipper into the issue and found a few missing overflow checks. As well as defining a max gas that would allow for a big enough memory that would trigger the overflows.

This PR includes:
- move `SizeInWords` function to `tosca/utils` package so that it can be used by both ct and lfvm
- rename of `st.MaxGas` to `st.MaxGasUsedByCt`
- limit use `st.MaxGasUsedByCt` as upper bound in `gasDomain`.
- add `st.MaxMemoryExpansionSize` to memory parameters. also remove non interesting parameters to reduce number of generated cases.
- add specification test, testing rules sensitive to this overflow
- add usages of `tosca.SizeInWords` and `sumWithOverflow` to specification in vulnerable effects.
- add usages of `tosca.SizeInWords` in lfvm instructions where applicable.
- add test to `lfvm/instructions_tests.go` to check sensitive rules for these overflow issue.
- add gas overflow check to call operations. 

This PR fixes #524
